src/ORM/Hierarchy/MarkedSet.php

```diff
@@ -413,7 +413,7 @@ protected function getSubtree($node, $depth = 0)
             'expanded' => $expanded,
             'opened' => $opened,
             'depth' => $depth,
-            'count' => $numChildren, // Count of DB children
+            'count' => ($limited && $numChildren > $this->getMaxChildNodes()) ? 0 : $numChildren, // Count of DB children
```

Comment:
I'd rather see this ternary outside the array, PHPStorm marks it as "confusing statement" as well, maybe wrap the second part in brackets?